internal/gatewayapi/securitypolicy.go

```diff
@@ -1507,7 +1512,25 @@ func (o *OpenIDConfig) validate() error {
 	return nil
 }
 
-func fetchEndpointsFromIssuer(issuerURL string, providerTLS *ir.TLSUpstreamConfig) (*OpenIDConfig, error) {
+func (t *Translator) fetchEndpointsFromIssuer(issuerURL string, providerTLS *ir.TLSUpstreamConfig) (*OpenIDConfig, error) {
+	if config, cachedErr, ok := t.oidcDiscoveryCache.Get(issuerURL); ok {
+		if cachedErr != nil {
+			return nil, cachedErr
+		}
+		return config, nil
+	}
+
+	config, err := discoverEndpointsFromIssuer(issuerURL, providerTLS)
```

Comment:
any estimation of size of this cache ?

do we need to implement eviction ?

Reply:
cool caching per translation solves the immediate need w/o the complexity of invalidation, thanks